### PR TITLE
fix(notifications): Assigned Notification Title

### PR DESCRIPTION
**Pull request by mgaeta**
The "assigned" notification always says that the issues were assigned by the assignee. This PR consolidates the email and slack messages.

### CODE REVIEW

File: src/sentry/notifications/notifications/activity/assigned.py [hunk 1]
```diff
@@ -2,102 +2,74 @@
...
+def _get_user_option(assignee_id: int) -> User | None:
+    try:
+        return User.objects.get_from_cache(id=assignee_id)
+    except User.DoesNotExist:
+        return None
+
+
+def _get_team_option(assignee_id: int, organization: Organization) -> Team | None:
+    try:
+        return Team.objects.get(id=assignee_id, organization=organization)
```
Comment (scefali) [inline src/sentry/notifications/notifications/activity/assigned.py:19]: @mgaeta Nit: you could just do `Team.objects.filter().first` and avoid having to use the try/except

File: src/sentry/notifications/notifications/activity/assigned.py [hunk 2]
```diff
@@ -2,102 +2,73 @@
...
+def get_assignee_str(activity: Activity, organization: Organization) -> str:
+    """Get a human-readable version of the assignment's target."""
+
+    assignee_id = activity.data.get("assignee")
+    assignee_type = activity.data.get("assigneeType", "user")
+    assignee_email = activity.data.get("assigneeEmail")
+
+    if assignee_type == "user":
+        if activity.user_id == assignee_id:
+            return "themselves"
```
[the remark this answers is not in the document]
Comment (mgaeta) [reply src/sentry/notifications/notifications/activity/assigned.py:31]: No they wouldn't be used anywhere else. All of the unit tests are for `get_description()` and `get_notification_title()`.

**Review by NisanthanNanthakumar — APPROVED** [review]

File: src/sentry/notifications/notifications/activity/assigned.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (scefali) [reply src/sentry/notifications/notifications/activity/assigned.py:31]: Can we make it an internal constant at least?

**Review by scefali — APPROVED** [review]
@mgaeta LGTM but making that change for making constants for the string literals would be nice

**Pull request merged**